### ISSUE TITLE
feat: implement Supabase sync queue processing

### ISSUE THREAD

**Issue by kocan** [kocan/jobrun#47]
## Context
The offline-first infrastructure is well-designed:
- All tables have `sync_status` columns ('pending' / 'synced')
- `syncQueue.ts` has helper functions for counting and marking records
- `deleted_at` soft-delete columns support sync of deletions

But `processSyncQueue()` is a **complete stub** that does nothing:
```ts
export async function processSyncQueue(): Promise<{ synced: number; failed: number }> {
  // TODO: Implement actual Supabase sync
  return { synced: 0, failed: pending };
}
```

## What Needs to Be Built
1. **Push queue processor** — iterate pending records, upsert to Supabase
2. **Pull sync** — fetch server-side changes and merge into local DB
3. **Conflict resolution** — server-wins or last-write-wins strategy
4. **Retry with backoff** — add `sync_retry_count` and `sync_error` columns
5. **Trigger on connectivity restore** — use NetworkProvider to auto-sync when back online
6. **Background sync** — periodic sync while app is in foreground

## Schema Changes Needed
```sql
ALTER TABLE customers ADD COLUMN sync_retry_count INTEGER DEFAULT 0;
ALTER TABLE customers ADD COLUMN sync_error TEXT;
-- (repeat for all syncable tables)
```

## Supabase Tables Needed
Mirror the local schema in Supabase with RLS policies per user.

## Severity
🟠 High — without sync, device loss = total data loss

## From
Code review findings F-14, F-16
